library/modules/EventManager.cpp

```diff
@@ -75,8 +88,9 @@ int32_t DFHack::EventManager::registerTick(EventHandler handler, int32_t when, P
                 Core::getInstance().getConsole().print("EventManager::registerTick: warning! absolute flag=false not honored.\n");
         }
     }
-    handler.freq = when;
-    tickQueue.insert(pair<int32_t, EventHandler>(handler.freq, handler));
+    handler.when = when;
+    tickQueue.insert(pair<int32_t, EventHandler>(handler.when, handler));
+    eventLastTick[handler.eventHandler] = -1;
     handlers[EventType::TICK].insert(pair<Plugin*,EventHandler>(plugin,handler));
     return when;
 }
```

Comment:
Perhaps we shouldn't add tick listeners to the `handlers[TICK]` container. The idea here is that it doesn't provide anything useful to tick events, it is just another thing to clean up inside of `manageTickEvent` in fact.